src/socket.cr

```diff
@@ -482,7 +482,7 @@ class Socket < IO
   end
 
   def fcntl(cmd, arg = 0)
```

Comment:
The `Socket#fcntl` and `Socket.fcntl` methods aren't portable, rely on LibC constants to call. We might want to consider deprecating 'em?